docs/reference/modules/cluster/remote-clusters-api-key.asciidoc

```diff
@@ -0,0 +1,177 @@
+production environment.
+
+.. Unzip the generated `cross-cluster-ca.zip` file. This compressed file
+contains the following content:
++
+[source,txt]
+----
+/ca
+|_ ca.crt
+|_ ca.key
+----
+
+.. Generate a certificate and private key pair for the nodes in the remote 
+cluster:
++
+[source,sh]
+----
+./bin/elasticsearch-certutil cert --out=cross-cluster.p12 --pass=CERT_PASSWORD --ca-cert=ca/ca.crt --ca-key=ca/ca.key --ca-pass=CA_PASSWORD --dns=example.com --ip=127.0.0.1
+----
++
+* Replace `CA_PASSWORD` with the CA password from the previous step.
+* Replace `CERT_PASSWORD` with the password you want to use for the generated
+private key.
+* Use the `--dns` option to specify the relevant DNS name for the certificate.
+You can specify it multiple times for multiple DNS.
+* Use the `--pi` option to specify the relevant IP address for the certificate.
+You can specify it multiple times for multiple IP addresses.
+
+.. If the remote cluster has multiple nodes, you can either:
++
+* create a single wildcard certificate for all nodes;
+* or, create separate certificates for each node either manually or in batch
+with the <<certutil-silent,silent mode>>.
+
+. On every node of the remote cluster:
+.. Copy the `cross-cluster.p12` file from the earlier step to the `config`
+directory. If you didn't create a wildcard certificate, make sure you copy the
+correct node-specific p12 file.
+.. Add following configuration to `elasticsearch.yml`:
++
+[source,yaml]
+----
+xpack.security.remote_cluster_server.ssl.enabled
+xpack.security.remote_cluster_server.ssl.keystore.path=cross-cluster.p12
+----
+
+.. Add the SSL keystore password to the {es} keystore:
++
+[source,sh]
+----
+./bin/elasticsearch-keystore add xpack.security.remote_cluster_server.ssl.keystore.secure_password
+----
++
+When prompted, enter the `CERT_PASSWORD` from the earlier step.
+
+. Restart the remote cluster.
+
+. On the remote cluster, generate a cross-cluster API key using the
+<<security-api-create-cross-cluster-api-key>> API or
+{kibana-ref}/api-keys.html[Kibana]. Grant the key the required access for {ccs}
+or {ccr}.
+
+===== On the local cluster
+
+. On every node of the local cluster:
+
+.. Copy the `ca.crt` file generated on the remote cluster earlier into the
+`config` directory, renaming the file `remote-cluster-ca.crt`.
+
+.. Add following configuration to `elasticsearch.yml`:
++
+[source,yaml]
+----
+xpack.security.remote_cluster_client.ssl.enabled: true
+xpack.security.remote_cluster_client.ssl.certificate_authorities: [ "remote-cluster-ca.crt" ]
+----
+
+.. Add the cross-cluster API key, created on the remote cluster earlier, to the
+keystore:
++
+[source,sh]
+----
+./bin/elasticsearch-keystore add cluster.remote.ALIAS.credentials
+----
++
+Replace `ALIAS` with the alias you will use to connect to the remote cluster 
+later. When prompted, enter the cross-cluster API key created on the remote
+cluster earlier.
```

Comment:
Either here or in the above where the cross-cluster API key is generated, we may want to say what is needed is the value of the `encoded` field from the CreateCrossApiKey response, e.g.
```
{
  "id": xxx,
   "api_key": xxx,
   "name": xxx,
   "encoded": "This_Is_What_We_Need"
}
```